src/api/Nhs.Appointments.Api/Notifications/BookingNotifier.cs

```diff
@@ -1,19 +1,46 @@
-﻿using Nhs.Appointments.Core;
+using Microsoft.Extensions.Logging;
+using Nhs.Appointments.Core;
+using Nhs.Appointments.Core.Messaging.Events;
 using System;
 using System.Collections.Generic;
+using System.Text.RegularExpressions;
 using System.Threading.Tasks;
 
 namespace Nhs.Appointments.Api.Notifications;
 
-public class BookingNotifier(ISendNotifications notificationClient, INotificationConfigurationStore notificationConfigurationStore, ISiteService siteService, IPrivacyUtil privacy) : IBookingMadeNotifier, IBookingReminderNotifier, IBookingCancelledNotifier, IBookingRescheduledNotifier
+public class BookingNotifier(
+    ISendNotifications notificationClient, 
+    INotificationConfigurationService notificationConfigurationService,
+    ISiteService siteService, 
+    IPrivacyUtil privacy,
+    ILogger<BookingNotifier> logger) : IBookingNotifier
 {
-    public async Task Notify(string eventType, string service, string bookingRef, string siteId, string firstName, DateOnly date, TimeOnly time, string email, string phoneNumber)
+    public async Task Notify(string eventType, string service, string bookingRef, string siteId, string firstName, DateOnly date, TimeOnly time, NotificationType notificationType, string destination)
     {
-        var site = await siteService.GetSiteByIdAsync(siteId);
+        if(notificationType == NotificationType.Unknown)
+        {
+            logger.LogWarning($"Unable to send notification as event type is unknown");
+            return;
+        }
 
+        if (ValidateDestination(notificationType, destination) == false)
+        {            
+            logger.LogWarning($"Unable to send notification for {eventType}:{bookingRef} to {notificationType}:{destination} because the destination is invalid");
+            return;
+        }
```

Comment:
Happy to leave this as it is, but I can't help notice this pattern occurs a lot and I wonder if it might be cleaner to introduce an `IValidator<PatientBookingNotificationEventBase>` (or similar, just whatever the root Notification object is). I wonder if you could then abstract away the "Unknown notification type" logic? 

(though I also notice each method takes a variety of parameters so maybe it wouldn't be as clean as I'm imagining - it's just a thought)